python/flow_sdk/shim_airbyte_cdk.py

```diff
@@ -24,15 +29,48 @@
 
 DOCS_URL = os.getenv("DOCS_URL")
 
+
+"""
+Airbyte doesn't appear to reduce state like we do. As a result,
+the Airbyte CDK is expecting its state input to look like a list of
+AirbyteStateMessages. Since we _do_ reduce state, what we do here is
+keep track of the latest AirbyteStateMessage for each stream, and then
+just give Airbyte a list of all of the latest state messages on boot.
+"""
+@dataclass
+class State:
+    stream_state: t.Optional[t.Dict[str, AirbyteStateMessage]] = None
+    global_state: t.Optional[AirbyteStateMessage] = None
+
+    def handle_message(self, msg: AirbyteStateMessage):
+        if msg.type == AirbyteStateType.STREAM:
+            if msg.stream == None:
+                raise Exception(
+                    "Got a STREAM-specific state message with no stream-specific state"
+                )
+            if not self.stream_state:
+                self.stream_state = {}
+            self.stream_state[
+                f"{msg.stream.stream_descriptor.name}-{msg.stream.stream_descriptor.namespace}"
+            ] = msg
+        elif msg.type == AirbyteStateType.GLOBAL:
+            self.global_state = msg
+        elif msg.type == AirbyteStateType.LEGACY or msg.type is None:
+            raise Exception("Airbyte LEGACY state is not supported")
```

Comment:
I have no idea, and it seemed like the docs around legacy state were saying "here there be dragons", so I'd rather have it be a clearly defined crash that we can add support for once we find a connector that wants it, rather than tracking down weird behavior that stems from unexpectedly formatted state messages